tests/proxy.rs

```diff
@@ -20,3 +23,33 @@ async fn test_lifecycle() {
         .expect("app shuts down")
         .expect("app exits without error")
 }
+
+#[tokio::test]
+async fn test_quit_lifecycle() {
+    // need to wait for the previous test release the port resource
+    thread::sleep(Duration::from_secs(1));
```

Comment:
This is a sign we are not actually properly shutting down..we should fix this. Ok for now though